DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/GroupsServiceTests.cs

```diff
@@ -522,12 +786,100 @@ private void SetUpAddCourseEnrolProcessFakes(GroupCourse groupCourse)
+                    .Returns(customPrompts);
+            }
+
+            A.CallTo(() => groupsDataService.GetGroupsForCentre(A<int>._)).Returns(groupsAtCentre ?? new List<Group>());
+
+            A.CallTo(() => groupsDataService.AddDelegateGroup(A<GroupDetails>._)).Returns(newGroupId);
+
+            A.CallTo(
+                    () => groupsDataService.AddDelegatesWithMatchingAnswersToGroup(
+                        A<int>._,
+                        A<int>._,
+                        A<int>._,
+                        A<string>._,
+                        null
+                    )
+                )
+                .DoesNothing();
+        }
+
+        private void AssertCorrectMethodsAreCalledForGenerateGroups(
+            GroupGenerationDetails groupGenerationDetails,
+            DateTime timeNow,
+            int linkedToField,
+            string? fullGroupLabel,
+            string? customPromptOptionText = null,
+            int? jobGroupId = null
+        )
+        {
+            var isJobGroup = groupGenerationDetails.RegistrationField.Equals(RegistrationField.JobGroup);
+
+            using (new AssertionScope())
+            {
+                if (!isJobGroup)
+                {
+                    A.CallTo(
+                            () => centreCustomPromptsService.GetCustomPromptsThatHaveOptionsForCentreByCentreId(
+                                groupGenerationDetails.CentreId
+                            )
+                        )
+                        .MustHaveHappenedOnceExactly();
+                    A.CallTo(() => groupsDataService.GetGroupsForCentre(groupGenerationDetails.CentreId))
+                        .MustHaveHappenedOnceExactly();
+                }
+
+                A.CallTo(
+                    () => groupsDataService.AddDelegateGroup(
+                        A<GroupDetails>.That.Matches(
+                            gd =>
+                                gd.CentreId == groupGenerationDetails.CentreId &&
+                                gd.GroupLabel == fullGroupLabel &&
+                                gd.GroupDescription == null &&
+                                gd.AdminUserId == groupGenerationDetails.AdminId &&
+                                gd.CreatedDate == timeNow &&
+                                gd.LinkedToField == linkedToField &&
+                                gd.SyncFieldChanges == groupGenerationDetails.SyncFieldChanges &&
+                                gd.AddNewRegistrants == groupGenerationDetails.AddNewRegistrants &&
+                                gd.PopulateExisting == groupGenerationDetails.PopulateExisting
+                        )
+                    )
+                ).MustHaveHappenedOnceExactly();
+
+                A.CallTo(
+                        () => groupsDataService.AddDelegatesWithMatchingAnswersToGroup(
+                            1,

```

Comment:
Do we want to define this as the newGroupId somewhere? Given that it is used in  `SetUpGenerateGroupFakes` then maybe making it a private const will do?

Reply:
Good idea, defined as a private const